### ISSUE TITLE
Suppress Line-Length violations for text-block content: need better approach than Regex

### ISSUE THREAD

**Issue by mohitsatr** [checkstyle/checkstyle#17707]
from: https://github.com/checkstyle/checkstyle/pull/17666#discussion_r2307295384

After #17666, LineLength Check will stop working on lines in the Text Blocks, but there are some edge cases that need special attention and probably a better approach than Regex to handle them.

Code:
```java
/*** some javadoc. */
public class Example1 {

  void testMethod1() {
    String s1 =
        """
        Contrary to popular belief, Lorem Ipsum is not simply random text. It has roots in a piece of classical
        """
           + getSampleTest();

    String s2 =
        """
        Contrary to popular belief, Lorem Ipsum is not simply random text. It has roots in a piece of classical""";

    String s3 = "Contrary to popular belief, Lorem Ipsum is not simply random text. It has roots in a piece of classical";
    // false-negative above

  String getSampleTest() {
    return "String";
  }
}
```
CI:
```
$ java -jar checkstyle-10.26.1-all.jar -c google_checks.xml Example1.java 
Starting audit...
Audit done.
```

But if the normal string `s3` is placed above the text blocks, we do get the violation.

Code:
```java
/*** some javadoc. */
public class Example2 {

  void testMethod1() {
    String s3 = "Contrary to popular belief, Lorem Ipsum is not simply random text. It has roots in a piece of classical";
    // ok above

    String s1 =
        """
        Contrary to popular belief, Lorem Ipsum is not simply random text. It has roots in a piece of classical
        """
           + getSampleTest();

    String s2 =
        """
        Contrary to popular belief, Lorem Ipsum is not simply random text. It has roots in a piece of classical""";
  }

  String getSampleTest() {
    return "String";
  }
}
```

CI:
```
$ java -jar checkstyle-10.26.1-all.jar -c google_checks.xml Example2.java 
Starting audit...
[WARN] /mnt/5D92528E6B945467/test/testing/Example2.java:5: Line is longer than 100 characters (found 122). [LineLength]
Audit done.
```

**Reason**-The following Suppression is used to turn-off LineLength Check from checking text-blocks:
```
  <-- Suppress to disable Check for Text-blocks -->
  <module name="SuppressWithPlainTextCommentFilter">
    <property name="checkFormat" value="LineLength"/>
    <property name="offCommentFormat" value='^.*"""\s*$'/>
    <property name="onCommentFormat" value='^\s*"""\s*(?:[,;]|.+)$'/>
  </module>
```

Taking the `s1` from `Example1` :
```
    String s1 =
        """
        Contrary to popular belief, Lorem Ipsum is not simply random text. It has roots in a piece of classical
        """
          + getSampleTest();
```

The opening quotes are on new line, so they matche the `offCommentFormat` regex pattern. So from this point Suppression is activated.
For the closing quotes after the content, Ideally `onCommentFormat` regex pattern would match string, but this closing quotes are also on new line, and not followed with `;` or `+`. In other words, they are identically to opening quotes. So even though they does match `onCommentFormat` regex pattern, `offComment` property overrides it. (because in the logic `if`-`if` is used) So Suppression is still activated.

https://github.com/checkstyle/checkstyle/blob/87b7306edb88081276dbab80d9afed7819186172/src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressWithPlainTextCommentFilter.java#L277-L284

The above process follows for the next code in the file, `onCommentFormat` always gets overriden by `offCommentFormat`, Suppression is never turned-on, disabling the LineLength Check in the file.

<img width="1162" height="468" alt="Image" src="https://github.com/user-attachments/assets/fe650fc1-81b4-4092-bf0b-a0ab32d56af7" />

As a part of the fix, we need to figure out how to handle these edge cases.

**Comment by romani** [comment]
> because in the logic if-if is used) So Suppression is still activated.

this logic is correct and it designed for different type comments, for `// BEGIN something`  and `// END something`.
So better to not hack SuppressWithPlainTextCommentFilter and keep it as is.

For all edge cases we need we need to create new Check , something like JavaLineLengh (imaginary Check) that can allow violations under some tokens.
Such new Check might be javadoc aware too, to allow allowances under some javadoc specific tokens, but javadoc Checks has performance problems, so we might delay doing this and rely on suppression.

First version of Check can be without any properties and no javadoc support and rely on suppression Xpath module to help filterout violation by location at/under specific AST node.

**Comment by romani** [comment]
@mohitsatr , @Zopsss  , do you agree on JavaLineLengh ?

**Comment by mohitsatr** [comment]
@romani yes I agree. we definitely need a new check that can be triggered by removing/adding Tokens.

**Comment by pree-T** [comment]
Hi, I want to work on this issue, how to start contributing?

**Comment by romani** [comment]
@pree-T ,
Please [read](https://github.com/checkstyle/checkstyle/blob/master/.github/CONTRIBUTING.md#getting-started) and watch videos at [Starting_Development](https://checkstyle.org/beginning_development.html#Starting_Development).
We highly recommend to start with simple issues first.

This issue is not simple, you can come back to it in few weeks.

**Comment by DakshRJain737** [comment]
I am on this issue. Already started working.
